web/pdf_thumbnail_view.js

```diff
@@ -405,11 +423,13 @@ var PDFThumbnailView = (function PDFThumbnailViewClosure() {
     },
   };
 
+  PDFThumbnailView.cleanup = function() {
```

Comment:
I'm a bit confused about why we need this construction (and why we even have it at the moment). How I see it is that `PDFThumbnailViewer` manages all `PDFThumbnailView` instances. Wouldn't it be much easier and more logical to move this cleanup logic to `PDFThumbnailViewer` instead? It would go over all `PDFThumbnailView` instances, destroy them and when they are all destroyed, call the factory's `destroyCanvas` method to delete the shared canvas.